x-pack/plugins/maps/public/shared/layers/sources/source.js

```diff
@@ -32,6 +33,10 @@ export class AbstractSource {
     return [];
   }
 
+  getInspectorAdapters() {
+    return this._inspectorAdapters;
+  }
```

Comment:
So we need to introduce this method? Could a layer-instance keep a reference to the `_inspectorAdapters`. 

I'd add `this._inspectorAdapters = inspectorAdapters` to the `AbstractLayer` constructor, which takes a second argument, just like AbstractSource

Not 100% sure, but I wouldn't expand the interface of sources, just so we can pull a reference to the inspectoradapter back up in the layers.

Reply:
Summing up our chat, I am going to leave as is since Layer does not need access to inspectorAdapters except in one isolated condition - when the VectorLayer creates LeftInnerJoin. In this case, VectorLayer never uses inspectorAdapters, it just needs access to inspectorAdapters to pass to the LeftInnerJoin constructor.

Adding inspectorAdapters to Layer constructor would spider web out since layers are created all over the place